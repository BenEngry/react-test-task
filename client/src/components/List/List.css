.list__item:not(:last-child) {
    margin-bottom: 30px;
}

.list__item {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    border: 2px solid #fff;
    border-radius: 20px;
    padding: 10px 20px;
    transition: .3s ease-in-out;
}

.list__string {
    position: relative;
}

.list__string::after {
    position: absolute;
    content: "";
    width: 2px;
    height: 99px;
    color: rgb(106, 62, 189);
    background-color: rgb(106, 62, 189);
    left: 115%;
    top: -122%;
    transition: .4s ease-in-out;
}

.list__string:hover::after {
    color: rgb(48, 18, 105);
    background-color: rgb(48, 18, 105);
}

.list__item:hover {
    transform: translateY(-10px);
    background-color: #fff;
    font-weight: 600;
    color: rgb(48, 18, 105);
}