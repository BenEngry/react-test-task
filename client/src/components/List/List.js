import { React } from "react";
import { useState, useEffect } from "react";
import { Spinner } from "../Spinner/Spinner";
import { ErrorMessage } from "../Error/ErrorMessage";
import { DATA_URL } from "../utils/consts";
import "./List.css";

export const List = () => {
    const [error, setError] = useState(null);
    const [isLoaded, setIsLoaded] = useState(false);
    const [items, setItems] = useState([]);

    useEffect(() => {
        fetch(DATA_URL)
            .then((response) => response.json())
            .then(
                (result) => {
                setIsLoaded(true);
                setItems(result.items);
            },
            (error) => {
                setIsLoaded(true);
                setError(error);
            }
        );
    }, []);

    if (error) {
        return <ErrorMessage error={error.message} />;
    } else if (!isLoaded) {
        return <Spinner />;
    } else {
        return (
        <div className="wrapper">
            {items.map((item) => (
            <div key={item.id} className="list__item">
                <a className="list__link" href={`/list/get/${item.id}`}>
                {item.name}
                </a>
            </div>
            ))}
        </div>
        );
    }
};
