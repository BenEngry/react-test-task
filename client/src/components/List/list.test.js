import renderer from "react-test-renderer";
import React from "react";
import * as matchers from 'jest-extended';
import { List } from "./List";
expect.extend(matchers);


describe("Testing <List/>", ()=> {

    it("List have rendered correctly", ()=>{
        const list = renderer.create(<List/>).toJSON();
        expect(list).toMatchSnapshot();
    });

    it("List state 'error' is empty object", ()=>{
        const error = renderer.create(<List/>).toJSON();
        expect(error.state().error).toBeObject();
    });
    
});
