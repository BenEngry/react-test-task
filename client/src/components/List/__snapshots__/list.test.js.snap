// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <List/> List have rendered correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader"
  >
    Loading...
  </div>
</div>
`;
